starlette/responses.py

```diff
@@ -342,5 +379,154 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
                             "more_body": more_body,
                         }
                     )
-        if self.background is not None:
-            await self.background()
+
+    async def _handle_single_range(
+        self, send: Send, start: int, end: int, file_size: int, send_header_only: bool
+    ) -> None:
+        self.headers["content-range"] = f"bytes {start}-{end - 1}/{file_size}"
+        self.headers["content-length"] = str(end - start)
+        await send({"type": "http.response.start", "status": 206, "headers": self.raw_headers})
+        if send_header_only:
+            await send({"type": "http.response.body", "body": b"", "more_body": False})
+        else:
+            async with await anyio.open_file(self.path, mode="rb") as file:
+                await file.seek(start)
+                more_body = True
+                while more_body:
+                    chunk = await file.read(min(self.chunk_size, end - start))
+                    start += len(chunk)
+                    more_body = len(chunk) == self.chunk_size and start < end
+                    await send({"type": "http.response.body", "body": chunk, "more_body": more_body})
+
+    async def _handle_multiple_ranges(
+        self,
+        send: Send,
+        ranges: list[tuple[int, int]],
+        file_size: int,
+        send_header_only: bool,
+    ) -> None:
+        boundary = "".join(random_choices("abcdefghijklmnopqrstuvwxyz0123456789", k=13))
+        content_length, header_generator = self.generate_multipart(
+            ranges, boundary, file_size, self.headers["content-type"]
+        )
+        self.headers["content-range"] = f"multipart/byteranges; boundary={boundary}"
+        self.headers["content-length"] = str(content_length)
+        await send({"type": "http.response.start", "status": 206, "headers": self.raw_headers})
+        if send_header_only:
+            await send({"type": "http.response.body", "body": b"", "more_body": False})
+        else:
+            async with await anyio.open_file(self.path, mode="rb") as file:
+                for start, end in ranges:
+                    await file.seek(start)
+                    chunk = await file.read(min(self.chunk_size, end - start))
```

Comment:
Does it send all content within the range?

Reply:
I'm not sure I understand the question...

I guess the answer would be "yes"?

Reply:
IIUC, this statement may send chunk with self.chunk_size, not the full content between `end - start`, and I didn't find similar logic to that in `handle_single_range`(pasted below) to loop until no more body to send：

```python
while more_body:
    chunk = await file.read(min(self.chunk_size, end - start))
    start += len(chunk)
    more_body = len(chunk) == self.chunk_size and start < end
    await send({"type": "http.response.body", "body": chunk, "more_body": more_body})
```

Reply:
I think we may benefit from some comments/docstrings around here. But I don't see anything wrong?

Could you try to come up with a test scenario that you think it would be problematic here?

This is a basic test for multiple ranges (to help):

```py
def test_file_response_merge_ranges(file_response_client: TestClient) -> None:
    response = file_response_client.get("/", headers={"Range": "bytes=0-100, 50-200"})
    assert response.status_code == 206
    assert response.headers["content-length"] == "201"
    assert response.headers["content-range"] == f"bytes 0-200/{len(README.encode('utf8'))}"
```

Reply:
Oh... I see your point.

Reply:
PR is welcome here, otherwise I'll work on it later.

Reply:
Thanks @frostming 🙏